packages/dcos-integration-test/extra/conftest.py

```diff
@@ -376,31 +377,29 @@ def get_test_app_in_docker(self, ip_per_container):
         return app, test_uuid
 
     def deploy_test_app_and_check(self, app, test_uuid):
-        service_points = self.deploy_marathon_app(app)
-        r = requests.get('http://{}:{}/test_uuid'.format(service_points[0].host,
-                                                         service_points[0].port))
-        if r.status_code != 200:
-            msg = "Test server replied with non-200 reply: '{0} {1}. "
-            msg += "Detailed explanation of the problem: {2}"
-            pytest.fail(msg.format(r.status_code, r.reason, r.text))
-
-        r_data = r.json()
+        with self.marathon_deploy_and_cleanup(app) as service_points:
+            r = requests.get('http://{}:{}/test_uuid'.format(service_points[0].host,
```

Comment:
nit / note for future cleanup: Some sort of requests wrapper where I could do a `.get('/test_uuid') would be cool for this.